### ISSUE TITLE
conv_transpose1d meta implementation allows invalid output_padding that real CUDA execution rejects

### ISSUE THREAD

**Issue by Isa-Fay** [pytorch/pytorch#178125]
## Summary
`torch.conv_transpose1d` validates `output_padding` on real execution, but the `meta` device path accepts the same invalid configuration and returns a shape.

## Reproduction
```python
import torch
import numpy as np
import torch
import torch.nn.functional as F

def call_func(inputs, stride=1, padding=0, output_padding=0, groups=1, dilation=1):
    if len(inputs) == 2:
        input_tensor, weight = inputs
        bias = None
    elif len(inputs) == 3:
        input_tensor, weight, bias = inputs
    else:
        raise ValueError("inputs must contain 2 or 3 tensors")
    
    return torch.conv_transpose1d(
        input=input_tensor,
        weight=weight,
        bias=bias,
        stride=stride,
        padding=padding,
        output_padding=output_padding,
        groups=groups,
        dilation=dilation
    )

# Test input that causes the defect
input_tensor = torch.randn(20, 16, 50)
weight = torch.randn(16, 33, 5)
inputs = [input_tensor, weight]

# Test parameters
stride = 2
padding = [0]
output_padding = 2
groups = 2
dilation = 1

print("Testing torch.conv_transpose1d defect reproduction")

# Test 1: Dynamic output shape (direct call)
print("\n1. Dynamic output shape (direct call):")
try:
    dynamic_result = call_func(inputs, stride=stride, padding=padding, 
                              output_padding=output_padding, groups=groups, dilation=dilation)
    print(f"Dynamic shape: {dynamic_result.shape}")
except Exception as e:
    print(f"Dynamic error: {e}")

# Test 2: Static output shape (torch.compile with dynamic=True)
print("\n2. Static output shape (torch.compile with dynamic=True):")
try:
    compiled_func = torch.compile(call_func, dynamic=True)
    static_result = compiled_func(inputs, stride=stride, padding=padding,
                                 output_padding=output_padding, groups=groups, dilation=dilation)
    print(f"Static shape: {static_result.shape}")
except Exception as e:
    print(f"Static error: {e}")

# Test 3: Meta output shape (device='meta')
print("\n3. Meta output shape (device='meta'):")
try:
    meta_input_tensor = torch.randn(20, 16, 50, device='meta')
    meta_weight = torch.randn(16, 33, 5, device='meta')
    meta_inputs = [meta_input_tensor, meta_weight]
    
    meta_result = call_func(meta_inputs, stride=stride, padding=padding,
                           output_padding=output_padding, groups=groups, dilation=dilation)
    print(f"Meta shape: {meta_result.shape}")
except Exception as e:
    print(f"Meta error: {e}")

print("\nDefect reproduced: Inconsistent behavior between dynamic/static execution (both fail with validation error) vs meta execution (succeeds and returns shape)")
```

## Expected behavior
Both real and meta execution should reject the invalid `output_padding` with the same error.

## Actual behavior
- Real CUDA/normal execution: raises
-   `output padding must be smaller than either stride or dilation`
- Meta execution: succeeds and returns `torch.Size([20, 66, 105])`

## Impact
This makes meta shape inference inconsistent with runtime validation and can let invalid models pass tracing/compilation checks.

### Versions

Collecting environment information...
PyTorch version: 2.10.0+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0
Clang version: Could not collect
CMake version: version 3.31.10
Libc version: glibc-2.35

Python version: 3.12.12 (main, Oct 10 2025, 08:52:57) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.6.113+-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           46 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  2
On-line CPU(s) list:                     0,1
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Xeon(R) CPU @ 2.20GHz
CPU family:                              6
Model:                                   79
Thread(s) per core:                      2
Core(s) per socket:                      1
Socket(s):                               1
Stepping:                                0
BogoMIPS:                                4399.99
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm rdseed adx smap xsaveopt arat md_clear arch_capabilities
Hypervisor vendor:                       KVM
Virtualization type:                     full
L1d cache:                               32 KiB (1 instance)
L1i cache:                               32 KiB (1 instance)
L2 cache:                                256 KiB (1 instance)
L3 cache:                                55 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0,1
Vulnerability Gather data sampling:      Not affected
Vulnerability Indirect target selection: Vulnerable
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Mitigation; PTE Inversion
Vulnerability Mds:                       Vulnerable; SMT Host state unknown
Vulnerability Meltdown:                  Vulnerable
Vulnerability Mmio stale data:           Vulnerable
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Vulnerable
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Vulnerable
Vulnerability Spectre v1:                Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:                Vulnerable; IBPB: disabled; STIBP: disabled; PBRSB-eIBRS: Not affected; BHI: Vulnerable
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Vulnerable
Vulnerability Vmscape:                   Not affected

Versions of relevant libraries:
[pip3] intel-cmplr-lib-ur==2025.3.2
[pip3] intel-openmp==2025.3.2
[pip3] mkl==2025.3.1
[pip3] numpy==2.0.2
[pip3] nvidia-nccl-cu12==2.29.7
[pip3] onemkl-license==2025.3.1
[pip3] optree==0.19.0
[pip3] tbb==2022.3.1
[pip3] tcmlib==1.4.1
[pip3] torch==2.10.0+cpu
[pip3] torchao==0.10.0
[pip3] torchaudio==2.10.0+cpu
[pip3] torchcodec==0.10.0
[pip3] torchdata==0.11.0
[pip3] torchsummary==1.5.1
[pip3] torchtune==0.6.1
[pip3] torchvision==0.25.0+cpu
[pip3] umf==1.0.3
[conda] Could not collect

cc @ezyang @eellison @bdhirsh @bobrenjc93 @aorenste

**Comment by holysheep123** [comment]
Hi **Isa-Fay** 鈥?"model not found" means one of:
1. Typo 鈥?double-check exact model name (e.g. claude-3-5-sonnet-latest)
2. Model not available on your plan/tier
3. Wrong API endpoint

What model name are you passing and which provider?

鈥?holysheep.ai
